### PR TITLE
docs: add common terms documentation to website

### PR DESCRIPTION
**Pull request by alwaysintreble**
@Ijwu @LegendaryLinux some website documentation. Let me know if there's anything you believe is missing.

### CODE REVIEW

**Comment by SoldierofOrder** [comment]
I have some suggestions for a few of the terms:

- I think we should reorder the first three terms to go Item -> Location -> Check instead of Check -> Item -> Location. The Check definition references locations, and the Location definition references Items, so a brand newbie might have an easier time parsing things if they read the Items definition first.

- I think for the Items definition, we should avoid mentioning the server (I totally recognize if y'all have a reason why we should mention it! I'm just under the impression that "item" can be defined without reference to online or multiplayer functionality). Here's my own attempt to rephrase: "Items are the things that get shuffled around (in other words, "randomized") as part of using a randomizer."
- For the Items definition, if we still want to mention something about how items work in Archipelago specifically, maybe we add something like "In Archipelago, your items aren't just randomized within your own game. If you've set up an Archipelago multiworld, all items are shuffled between all of the worlds."

- Maybe I'm off base here, but I feel like "check" as a _verb_ is very intuitive for native English speakers, e.g., "have you checked the basin fountain" makes sense for anyone who knows what "the basin fountain" means. I think the weirdness (and thus, the need to define it) comes from when people use it as a noun, e.g. "i gotta go back for the six checks I missed in Deepnest". I feel like this definition, as written, is trying to define it as a verb, so I think it's worth revising. I hadn't heard this word until I joined this community, like, a few days ago, so I'll leave my thoughts out on this one unless prompted, since I'm sure y'all have a better sense of what it means than I do.

I have a much weaker grasp on the other terms, so I'll come back to those if you want me to. :)

File: WebHostLib/static/assets/faq/common-terms_en.md
Comment (ThePhar) [inline WebHostLib/static/assets/faq/common-terms_en.md:21]: This is not entirely accurate, just because `yaml` setting files can have multiple yamls now. Maybe a small tweak to remove reference to `files`?

File: WebHostLib/static/assets/faq/common-terms_en.md
Comment (ThePhar) [inline WebHostLib/static/assets/faq/common-terms_en.md:52]: I'd probably add a note that something can be considered "out of logic" if accessing a location requires the use of special knowledge that the randomizer does not take into account when calculating if you can access the location.

**Review by ThePhar — COMMENTED** [review]
Couple key things that might be nice to tweak, in my humble opinion.

File: WebHostLib/static/assets/faq/common-terms_en.md
Comment (alwaysintreble) [reply WebHostLib/static/assets/faq/common-terms_en.md:21]: I think the original intent here I was planning on covering the case of using a weightedsettings.yaml but decided it was out of scope. will reevaluate this though

File: WebHostLib/static/assets/faq/common-terms_en.md
Comment (LegendaryLinux) [inline WebHostLib/static/assets/faq/common-terms_en.md:66]:
```md
## Burger King / BK Mode / BK
A term used to indicate when a player has no checks available to them, and is awaiting progression from another player. There are two types of BK Mode:  
1. "Logical BK" refers to when a player has no logical checks remaining, but does have the ability to acquire some checks out of logic.  
2. "Hard BK" refers to when a player has no checks remaining in or out of logic.

The origin of this term comes from a GDQ (Games Done Quick) event in which many players were participating in an Ocarina of Time MultiWorld. One player found themselves in hard BK mode, and decided to visit the Burger King attached to the hotel. That player returned and finished their meal before receiving a progression item.
```

File: WebHostLib/static/assets/faq/common-terms_en.md
Comment (LegendaryLinux) [inline WebHostLib/static/assets/faq/common-terms_en.md:4]: Line 4 is unnecessary. The title of the document indicates its purpose.

File: WebHostLib/static/assets/commonTerms.js
Comment (LegendaryLinux) [inline WebHostLib/static/assets/commonTerms.js:8]: "Sorry, the jargon file is not available in that language yet."

File: WebHostLib/static/assets/commonTerms.js
Comment (LegendaryLinux) [inline WebHostLib/static/assets/commonTerms.js:12]: "Something went wrong while loading the jargon file."

File: WebHostLib/__init__.py
Comment (LegendaryLinux) [inline WebHostLib/__init__.py:144]:
```py
@app.route('jargon/<string:lang>/')
```

**Review by LegendaryLinux — CHANGES_REQUESTED** [review]
I left a bunch of comments indicating requested changes to language. Overall this is a pretty good job. Thanks for writing this up!

File: WebHostLib/templates/siteMap.html
Comment (alwaysintreble) [reply WebHostLib/templates/siteMap.html:29]: I almost used the word Jargon which would be more "correct" but I figured it might confuse users who don't know what Jargon means so opted to use "common terms" as not necessarily everything documented here is jargon

**Comment by alwaysintreble** [comment]
i've made some changes and updates. would like to get @SoldierofOrder and @mDuo13 opinions on this

File: WebHostLib/static/assets/faq/common-terms_en.md
Comment (mDuo13) [inline WebHostLib/static/assets/faq/common-terms_en.md:23]: Maybe note that each player needs a unique name?

**Comment by LegendaryLinux** [comment]
I will not approve this until you use the word `jargon` as the name of your jargon file.

Your arguments against the word are:
1. Non-English speakers might not understand the word
2. Some English speakers might not understand the word

My responses are:
1. The page being loaded is the English version, as described in your route
2. We shouldn't just decide not to use words because people don't know what they mean. Some people don't know what an Archipelago is, but that's our name. Teach people, don't just leave them in ignorance.

File: WebHostLib/static/assets/faq/common-terms_en.md
Comment (mDuo13) [inline WebHostLib/static/assets/faq/common-terms_en.md:32]: I think this definition can be tightened up a bit to clarify the important parts more concisely. Here's an attempt:

```suggestion
World in terms of Archipelago can mean multiple things and is used interchangeably in many situations.
* During gameplay, a world is a single instance of a game, occupying one player "slot". However, 
Archipelago allows multiple players to connect to the same slot; then those players can share a world 
and complete it cooperatively. For games with native cooperative play, you can also play together and
share a world that way, usually with only one player connected to the multiworld.
* On the programming side, a world typically represents the package that integrates Archipelago with a
particular game. For example this could be the entire `worlds/factorio` directory.
```

File: WebHostLib/static/assets/faq/common-terms_en.md
Comment (mDuo13) [inline WebHostLib/static/assets/faq/common-terms_en.md:42]:
```suggestion
A "seed" is a number used to initialize a pseudorandom number generator. Whenever you generate a new game on Archipelago
this is a new "seed" as it has unique item placement, and you can create multiple "rooms" on the Archipelago site from a
single seed. Using the same seed results in the random placement being the same.
```

**Review by mDuo13 — APPROVED** [review]
Definitely a helpful improvement! I left some suggestions, but feel free to consider them optional.

**Comment by LegendaryLinux** [comment]
> to non-native speakers

This is the English version of the jargon file. Its hyperlink literally ends with `/en`

As for definition, let's just reference dictionary.com for this:  
https://www.dictionary.com/browse/jargon
> the language, especially the vocabulary, peculiar to a particular trade, profession, or group

> language that is characterized by uncommon or pretentious vocabulary and convoluted syntax and is often vague in meaning

https://www.dictionary.com/browse/glossary
> a list of terms in a special subject, field, or area of usage, with accompanying definitions

> such a list at the back of a book, explaining or defining difficult or unusual words and expressions used in the text

While glossary has a similar definition, their uses are distinct. A glossary is typically found at the end of a textbook or technical writing, while jargon references a set of common terms used in a specific field or group.

Jargon is the correct word for this file.

**Comment by mDuo13** [comment]
> > to non-native speakers
> 
> This is the English version of the jargon file. Its hyperlink literally ends with `/en`

People whose native language is not available may read the English version if their primary language is not available, which in most cases it is not. Personally, I'd rather provide a welcoming experience to a broader audience than gatekeep people based on their language proficiency.

> While glossary has a similar definition, their uses are distinct. A glossary is typically found at the end of a textbook or technical writing, while jargon references a set of common terms used in a specific field or group.

"Jargon" describes the words in the file. "Glossary" describes the file. Both are appropriate, but "glossary" is a better name for the file itself _and_ it's more commonly used in documentation, which this is. Also, [jargon file](https://en.wikipedia.org/wiki/Jargon_File) is the name of a specific notable publication, so unless you're trying to pay tribute to that instance, I'd say the word "glossary" is more natural.

That's the last I'll say on this. This PR represents real improvements on the beginner documentation and I'd rather not hold those up with an extended argument about semantics.

**Comment by SoldierofOrder** [comment]
I agree with @mDuo13 on these changes. This is a great foundation (for which I'm, of course, still going to have notes + suggestions :) but that comes later, once we have these approved), and I don't think it's worth delaying for this.

I'll weigh in on the discussion, though, in hopes of persuading @LegendaryLinux. First, on a philosophical note: we don't have extensive support for the documentation in all common languages, as I understand it, and the dominant language of our community is English. Therefore, people _will_ come to us only knowing a language we don't support, defaulting to English instead, and it's worth choosing language that's more accessible to them even when it conflicts with language that would suit our (very individual!) aesthetic sensibilities.

Then, on a definitional note: the Wikipedia page mDuo13 links to about the "Jargon File" even specifies that that file **is** a glossary. Based on the definitions you posted, I believe that anything that can be called a "jargon file" can also be called a glossary (even if the reverse is not necessarily true). So I see little reason to prefer one over the other if we're going purely based off of dictionary definitions.

For what it's worth, I approve this documentation as it stands, with the intention to suggest further changes later. :)

**Review by LegendaryLinux — CHANGES_REQUESTED** [review]
It seems like I'm in the minority. `Glossary` is better than `Common Terms`, at least. I'll approve once the page title and route are changed to `Glossary` and `/glossary/en` respectively.

**Comment by alwaysintreble** [comment]
> For what it's worth, I approve this documentation as it stands, with the intention to suggest further changes later. :)

Since this is open source and this document is just an md you can always just PR more stuff in yourself if you feel inclined. I'm not a super great writer so I have no issue with people editing my text as long as I don't feel that it makes the understanding more complicated.

**Review by LegendaryLinux — APPROVED** [review]

**Pull request merged**
